### ISSUE TITLE
compile time error

### ISSUE THREAD

**Issue by aysha-statvalu** [stephanrauh/ngx-extended-pdf-viewer#1013]
when I installed the latest version ^11.0.0-alpha.2 I got this error during compilation time.

export 'firstValueFrom' (imported as 'firstValueFrom') was not found in 'rxjs' (possible exports: ArgumentOutOfRangeError, AsyncSubject, BehaviorSubject, ConnectableObservable, EMPTY, EmptyError, GroupedObservable, NEVER, Notification, NotificationKind, ObjectUnsubscribedError, Observable, ReplaySubject, Scheduler, Subject, Subscriber, Subscription, TimeoutError, UnsubscriptionError, VirtualAction, VirtualTimeScheduler, animationFrame, animationFrameScheduler, asap, asapScheduler, async, asyncScheduler, bindCallback, bindNodeCallback, combineLatest, concat, config, defer, empty, forkJoin, from, fromEvent, fromEventPattern, generate, identity, iif, interval, isObservable, merge, never, noop, observable, of,
onErrorResumeNext, pairs, partition, pipe, queue, queueScheduler, race, range, scheduled, throwError, timer, using,
zip)

**Comment by stephanrauh** [comment]
You need a current version of rxjs. If that's a problem to you, I can switch to the old API - but that'd be a pity!

**Comment by cghislai** [comment]
needs angular@^13 with rxjs@^7. This should be listed in the peerDependencies.

**Comment by aysha-statvalu** [comment]
but angular 13 is new, I am using angular 12

**Comment by stephanrauh** [comment]
You're right. A few minutes ago, I also observed that `firstValueFrom()` isn't supported by Angular 12 out-of -the-box. I'll fix this.

**Comment by stephanrauh** [comment]
@cghislai Right you are! That's why I call so many versions "alpha". I want to get away with the occasional mistake. :) 

Be that as it may, I like your hint. It never crossed my mind I should mention the dependencies to RxJS and other libraries, but no doubt about that, it's something I should do!

**Comment by hulyak** [comment]
Hi, I can see that [PDFViewer.net](https://pdfviewer.net/extended-pdf-viewer/simple) is running on Angular 13. I'm trying to run with version 13, but I cannot see any PDF rendered on the browser. Is it compatible with version 13?

**Comment by stephanrauh** [comment]
Yes, I think so. As you've said, the showcase runs on Angular 13, and today I set up the pdf viewer three times in a row on Angular 13 to rehearse a presentation. 

It's possible you've run into an incompatibility I didn't notice yet. Which version of ngx-extended-pdf-viewer are you using?

**Comment by hulyak** [comment]
I'm using the `^11.0.0-alpha.2` version of `ngx-extended-pdf-viewer`. I'm getting this error with `ng serve` 

```

Error: node_modules/ngx-extended-pdf-viewer/lib/ngx-extended-pdf-viewer.component.d.ts:23:22 - error NG6002: Appears in the NgModule.imports of AppModule, but could not be resolved to an NgModule class.

This likely means that the library (ngx-extended-pdf-viewer) which declares NgxExtendedPdfViewerComponent has not been processed correctly by ngcc, or is not compatible with Angular Ivy. Check if a newer version of the library is available, and update if so. Also consider checking with the library's authors to see if the library is expected to be compatible with Ivy.

23 export declare class NgxExtendedPdfViewerComponent implements OnInit, AfterViewInit, OnChanges, OnDestroy {
                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by stephanrauh** [comment]
What a strange message! I've read the Angular 13 introduces changes for library authors. I just hope they didn't add a breaking change making it impossible to publish libraries that work with Angular 10, 11, 12, and 13. I'll have a look at it after work.

**Comment by cghislai** [comment]
@hulyak i think you are importing a component in your app module's `imports`. You need to import the module that provides the component (eg NgxExtendedPdfViewerModule), not the component itself. 

This is what i understand from `ngx-extended-pdf-viewer.component.d.ts:23:22 - error NG6002: Appears in the NgModule.imports of AppModule, but could not be resolved to an NgModule class.`

**Comment by hulyak** [comment]
Thanks, @cghislai. Yes, I was importing the component instead of the module. However, I can't still the PDF. I have the pdf file in the assets folder but it's showing a 404 error. I can only see the widgets at the top. I have double-checked the docs, but it's not showing the PDF.

<img width="710" alt="Screen Shot 2021-11-12 at 7 51 56 AM" src="https://user-images.githubusercontent.com/57311842/141495302-a7994d61-2182-4470-b660-dee6ed6e7a35.png">

**Comment by LAlves91** [comment]
Hey @hulyak !

Did you follow the [README instructions](https://github.com/stephanrauh/ngx-extended-pdf-viewer/tree/main/projects/ngx-extended-pdf-viewer#how-to-use-the-library)? These files are assets from this library, which you have to add to your own project assets (through angular.json asset configuration).

Cheers!

**Comment by aysha-statvalu** [comment]
will v11 only be available for angular 13 still i am getting same error in v12.

**Comment by LAlves91** [comment]
Hi @aysha-statvalu!

The error you mentioned is not really about Angular compatibility. It's because [rxjs v7](https://rxjs.dev/6-to-7-change-summary#firstvaluefrom) introduced the function `firstValueFrom()`, which is an improved version of `toPromise()`, which got deprecated.

According to [Angular Compatibility](https://gist.github.com/LayZeeDK/c822cc812f75bb07b7c55d07ba2719b3) information, it's possible to use rxjs v7 with Angular 12.2.x, so if you update your project's rxjs version, it should work.

Anyway, maybe it would be possible to check for rxjs version in ngx-extended-pdf-viewer (trying to import the function firstValueFrom dinamically, using toPromise otherwise). Do you think it's a good idea, @stephanrauh?

**Comment by hulyak** [comment]
Yes, I have added the assets inside `angular.json` file. @LAlves91 

```
    "assets": [
              "src/favicon.ico",
              "src/assets",
              {
                "glob": "**/*",
                "input": "node_modules/ngx-extended-pdf-viewer/assets/",
                "output": "/assets/"
              }
            ],
```

**Comment by LAlves91** [comment]
@hulyak weird...when I tested here, the assets were in place. Can you confirm the location your app is trying to load these files from?

**Comment by hulyak** [comment]
I just found it! I have added the config object for the `test`. I moved it under architect and build object. It's working now. I think you could make that clear in the README.

**Comment by stephanrauh** [comment]
@hulyak Phew... I'm glad you've found and solved the two mistakes. Your bug report was really worrying! :)

**Comment by stephanrauh** [comment]
@aysha-statvalu Don't worry, ngx-extended-pdf-viewer 11.0.0 is going to be compatible to Angular 12. More to the point, I'm trying to be compatible to the last four Angular versions. I can't promise this, because I don't know which breaking changes Angular introduces, but that's my goal. I'm positive I can support versions 10 through 13.

**Comment by stephanrauh** [comment]
Homework for myself: 

[ ] downgrade ngx-extended-pdf-viewer to RxJS 6 as long as it's compatible to the most recent version of RxJS.

**Comment by stephanrauh** [comment]
Solved with version 11.0.0-alpha.3.
